.github/workflows/pr_info_check.yml

```diff
@@ -0,0 +1,93 @@
+          flags: 's'
+          excludeDescription: 'true'
+          excludeTitle: 'false'
+          checkAllCommitMessages: 'false'
+      - name: Check PR description
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^(\[WIP\] )?([A-Z0-9]+-\d+) .{10,}$'
+          error: 'Pull Request name must have meaningful description'
+          flags: 's'
+          excludeDescription: 'true'
+          excludeTitle: 'false'
+          checkAllCommitMessages: 'false'
+      - name: Check commit message task board reference
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^D4CHAT-[0-9]+.*$'
+          error: 'Commit message first line must contain task reference.'
+          flags: 's'
+          excludeDescription: 'true'
+          excludeTitle: 'true'
+          checkAllCommitMessages: 'true'
+          accessToken: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check Commit Type
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '.* (feat|fix|docs|style|refactor|test|chore): .*$'
+          error: 'Commit message first line must contain one of the following commit types: feat, fix, docs, style, refactor, test, chore.'
+          flags: 's'
+          excludeDescription: 'true'
+          excludeTitle: 'true'
+          checkAllCommitMessages: 'true'
+          accessToken: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check Commit First Word Capitalization
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^[^:]+: [A-Z].*$'
+          flags: 's'
+          error: 'Commit message must start with a capital letter.'
+          excludeDescription: 'true'
+          excludeTitle: 'true'
+          checkAllCommitMessages: 'true'
+          accessToken: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check Commit First Line End
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^[^\n]+[^.\n](\n.+)*$'
+          flags: 's'
+          error: 'Commit message first line must not end with a period.'
+          excludeDescription: 'true'
+          excludeTitle: 'true'
+          checkAllCommitMessages: 'true'
+          accessToken: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check Commit Multiline Separator
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^[^\n]+(\n(\n[^\n]+)+)?$'
+          error: 'Commit header and body must be separated by one blank line'
+          flags: 'g'
+          excludeDescription: 'true'
+          excludeTitle: 'true'
+          checkAllCommitMessages: 'true'
+          accessToken: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check Commit Line Length
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^[^:]+: [^\n]{0,50}(\n[^\n]{0,72})*$'
+          error: 'The maximum line length of 50 (72) characters is exceeded.'
+          flags: 'g'
```

Comment:
Could you explain what purpose does flag serve? Thanks in advance

Reply:
The flag enables special instructions for regexs.

In our case we use two different flags:
- `g` (global) - don't return after the first match;
- `s` (single line) - dot matches newline;

You can find more information [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#advanced_searching_with_flags)